src/metabase/related.clj

```diff
@@ -173,32 +174,33 @@
 
 (defn- recently-modified-dashboards
   []
-  (->> (db/select-field :model_id 'Revision
-         :model     "Dashboard"
-         :user_id   api/*current-user-id*
-         {:order-by [[:timestamp :desc]]})
-       (map (partial db/select-one Dashboard :id))
-       filter-visible
-       (take max-serendipity-matches)))
+  (when-let [dashboard-ids (not-empty (t2/select-fn-set :model_id 'Revision
+                                                        :model     "Dashboard"
+                                                        :user_id   api/*current-user-id*
+                                                        {:order-by [[:timestamp :desc]]}))]
+    (->> (db/select Dashboard :id [:in dashboard-ids])
+         filter-visible
+         (take max-serendipity-matches))))
```

Comment:
This was icky because it was doing a `select-one` for every Dashboard ID rather than one single batched `select`. I didn't need to fix this, but I couldn't leave it like that in good conscience